src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -790,6 +794,22 @@ class module_sram : public module_impl
     bo.sync(XCL_BO_SYNC_BO_TO_DEVICE);
   }
 
+#ifdef _DEBUG
+  void
+  dump_bo(xrt::bo& bo, const std::string& filename)
+  {
+    if (xrt_core::config::get_feature_toggle(Debug_Bo_From_Elf_Feature)) {
+      std::ofstream ofs(filename, std::ios::out | std::ios::binary);
+
+      if (!ofs.is_open())
+        throw std::runtime_error("Failure opening file " + filename + " for writing!");
+
+      auto buf = bo.map<char*>();
+      ofs.write(buf, bo.size());
+    }
+  }
+#endif
```

Comment:
Move to global function under unnamed namespace within this compilation unit as there no need for state access.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c4-make-a-function-a-member-only-if-it-needs-direct-access-to-the-representation-of-a-class

Remove the surrounding `#ifdef _DEBUG` from the function.  It is okay and good to protect the call from create_instr_buf as you are doing.